src/libraries/System.Private.CoreLib/src/System/IO/Path.cs

```diff
@@ -926,5 +926,89 @@ private static string GetRelativePath(string relativeTo, string path, StringComp
         /// Returns true if the path ends in a directory separator.
         /// </summary>
         public static bool EndsInDirectorySeparator(string path) => PathInternal.EndsInDirectorySeparator(path);
+
+        /// <summary>
+        /// Removes redundant segments from the specified path string.
+        /// </summary>
+        /// <param name="path">The path to analyze.</param>
+        /// <returns>A string without redundant segments.</returns>
+        [return: NotNullIfNotNull("path")]
+        public static string? RemoveRedundantSegments(string? path)
+        {
+            if (path == null)
+                return null;
+
+            if (PathInternal.IsEffectivelyEmpty(path.AsSpan()))
+                return string.Empty;
+
+            ValueStringBuilder sb = new ValueStringBuilder(260 /* PathInternal.MaxShortPath */);
+
+            if (!RedundantSegmentHelper.TryRemoveRedundantSegments(path.AsSpan(), ref sb))
+            {
+                sb.Dispose();
+                return path;
+            }
+
+            return sb.ToString(); // Disposes
+        }
+
+        /// <summary>
+        /// Removes redundant segments from the specified path read-only span.
+        /// </summary>
+        /// <param name="path">The path to analyze.</param>
+        /// <returns>A string without redundant segments.</returns>
+        public static string RemoveRedundantSegments(ReadOnlySpan<char> path)
+        {
+            if (PathInternal.IsEffectivelyEmpty(path))
+                return string.Empty;
+
+            ValueStringBuilder sb = new ValueStringBuilder(260 /* PathInternal.MaxShortPath */);
+
+            string result;
```

Comment:
nit: You don't really need the result local. You can just return `path.ToString()` after disposing `sb` or `sb.ToString()`

Reply:
Fixed.